firebase-perf/src/main/java/com/google/firebase/perf/injection/components/FirebasePerformanceComponent.java

```diff
@@ -0,0 +1,31 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.perf.injection.components;
+
+import androidx.annotation.NonNull;
+import com.google.firebase.perf.FirebasePerformance;
+import com.google.firebase.perf.injection.modules.FirebasePerformanceModule;
+import dagger.Component;
+
+/**
+ * Dagger component to create FirebasePerformanceComponent Objects.
+ *
+ * @hide
+ */
+@Component(modules = {FirebasePerformanceModule.class})
+public interface FirebasePerformanceComponent {
+  @NonNull
+  FirebasePerformance getFirebasePerformance();
```

Comment:
Although currently `FirebasePerfRegistrar` is the only place where `FirebasePerformance` object is created so we should be fine but since Dagger allows us to define [Singleton](https://dagger.dev/dev-guide/) scope why not use that to guarantee singleton behavior?

Reply:
Good observation. We will introduce singleton with scopes in the next PR.